src/crimson/os/seastore/lba_manager/btree/btree_lba_manager.h

```diff
@@ -226,25 +218,61 @@ class BtreeLBAManager : public LBAManager {
       P_ADDR_ZERO,
       P_ADDR_NULL,
       L_ADDR_NULL,
+      0,
       nullptr);
   }
 
-  alloc_extent_ret clone_extent(
+  alloc_extent_ret clone_mapping(
     Transaction &t,
     laddr_t hint,
     extent_len_t len,
     laddr_t intermediate_key,
     paddr_t actual_addr,
-    laddr_t intermediate_base)
+    laddr_t intermediate_base,
+    extent_len_t intermediate_length)
   {
+    assert(intermediate_key != L_ADDR_NULL);
+    assert(intermediate_base != L_ADDR_NULL);
+    assert(intermediate_length != 0);
     return _alloc_extent(
       t,
       hint,
       len,
       intermediate_key,
       actual_addr,
       intermediate_base,
-      nullptr);
+      intermediate_length,
+      nullptr
+    ).si_then([&t, this, intermediate_base](auto indirect_mapping) {
+      assert(indirect_mapping->is_indirect());
+      return update_refcount(t, intermediate_base, 1, false
+      ).si_then([imapping=std::move(indirect_mapping), &t](auto p) mutable {
+	auto mapping = std::move(p.second);
+#ifndef NDEBUG
+	auto v = mapping->get_logical_extent(t);
```

Comment:
Hmm, how about introducing a **non-future** based `LBAMapping::is_stable() -> bool`?

My assumption is that:
* If the mapping is not linked to a child yet, it must be stable because it is absent in the cache.
* If the mapping is already linked to a child, we can peek whether the linked extent is stable, even though the it may still under io.

If my assumption is correct, the direct benefit here is that this section can be replaced by a straitforward ` assert(mapping->is_stable())`.

More importantly, seems to me there are use cases (should) rely on this interface to make decisions based on whether the pin is stable, e.g.:
* https://github.com/ceph/ceph/blob/main/src/crimson/os/seastore/object_data_handler.cc#L627-L640
* https://github.com/ceph/ceph/blob/main/src/crimson/os/seastore/transaction_manager.h#L348-L349
* https://github.com/ceph/ceph/pull/50419#discussion_r1378503243

Reply:
> * If the mapping is already linked to a child, we can peek whether the linked extent is stable, even though the it may still under io.

Do you mean that we treat a "MUTATION_PENDING" and "under-io" logical child extent as stable?

Reply:
> * If the mapping is already linked to a child, we can peek whether the linked extent is stable, even though the it may still under io.

We may find a `MUTATION_PENDING` extent when trying to `get_logical_extent`.

Reply:
> Do you mean that we treat a "MUTATION_PENDING" and "under-io" logical child extent as stable?

I think not, from `CachedExtent::is_stable()`, stable means it is either clean or dirty, which means the extent content hasn't been modified by the current transaction. Previously some code was checking this according to whether the paddr is absolute, but that's not general after RBM is introduced (RBM non-stable extents also have absolute paddrs).

Reply:
Sounds good! The discussion was that it is still possible to introduce `LBAMapping::is_stable() -> bool`.